### ISSUE TITLE
License question

### ISSUE THREAD

**Issue by alice-mkh** [hizzlekizzle/quark-shaders#12]
Hi, the readme says:

> licensed under the General Public License v2 

Does it mean GPL-2.0-only or GPL-2.0-or-later?

I noticed we use a shader (bicubic) marked as GPL-2.0+ while the readme seems to say otherwise, so I wonder if that's an oversight and we should remove it (the app is GPL-3.0-or-later so it's not compatible if the shaders are v2 only) or it's actually 2-or-later

**Comment by hizzlekizzle** [comment]
The shaders each have their own licenses. However, if the shader isn't being distributed as a compiled binary, you're just distributing a source file, which should be fine AFAIK. Of course, I'm not a lawyer, so don't interpret that as legal advice.

**Comment by alice-mkh** [comment]
I mean not all of them do - [that one](https://github.com/hizzlekizzle/quark-shaders/blob/master/Bicubic.shader/bicubic.fs) doesn't have a license specified, so it's using the one from readme.

**Comment by hizzlekizzle** [comment]
These shaders are (mostly) ported from other places. I need to go through and audit the licenses at some point.

For example, that one is a port of this guy (probably 3 hops worth of porting): https://github.com/libretro/common-shaders/blob/master/bicubic/shaders/bicubic-normal.cg

Luckily for you, it's GPLv2+ :)

**Comment by alice-mkh** [comment]
Yeah, I already did that ^^ (to this one: https://github.com/libretro/glsl-shaders/blob/master/cubic/shaders/bicubic.glsl) - it's more of a general question.

I was also thinking of using lcd-cgwg, but then I noticed there's a notice here, so that one should be good I think: https://github.com/libretro/common-shaders/blob/master/handheld/lcd-cgwg.cgp

**Comment by hizzlekizzle** [comment]
heh, license in the preset, jeeeeez

I think that one's just coming from the motionblur shader, but most of cgwg's contributions stick to GPLv2+, so you're probably safe there either way.
